build.py

```diff
@@ -34,28 +34,33 @@ def setup_jinja_environment(alerts):
     return env
 
 
-alerts = Alerts.from_yaml(REPO / 'data.yaml')
-env = setup_jinja_environment(alerts)
+def alerts_from_yaml():
+    alerts = Alerts.from_yaml(REPO / 'data.yaml')
+    return alerts
 
 
-if __name__ == '__main__':
-    ROOT.mkdir(exist_ok=True)
+def alerts_from_api():
+    raise NotImplementedError("This has not been implemented yet")
 
+
+def get_rendered_pages(alerts):
+    env = setup_jinja_environment(alerts)
+
+    rendered = {}
     for page in SRC.glob('*.html'):
         template = env.get_template(str(page))
 
+        # render each individual alert's page
         if str(page) == 'src/alert.html':
             for alert in alerts.public:
-                target = ROOT / alert.identifier
-                target.open('w').write(template.render({'alert_data': alert}))
+                rendered["alerts/" + alert.identifier] = template.render({'alert_data': alert})
             continue
 
         if 'index.html' in str(page):
-            target = DIST / page.relative_to(SRC)
+            rendered['alerts'] = template.render()
         else:
-            target = ROOT / page.relative_to(SRC)
+            target = str(page.relative_to(SRC))
+            target = target.replace(".html", "")
+            rendered["alerts/" + target] = template.render()
 
-        target.parent.mkdir(exist_ok=True)
-        target.open('w').write(template.render())
```

Comment:
Addressed both the local development point and the deployment point in https://github.com/alphagov/notifications-govuk-alerts/pull/151/commits/a04e66d43b74299c71d1c6d41335ad53f1e94903 and https://github.com/alphagov/notifications-govuk-alerts/pull/151/commits/e71fa9c0d523f932853d09f41ad020a6b7b01067 respectively.

Reply:
Thanks! Could we just keep the disk method for now? I was about to add some comments on a04e66d43b74299c71d1c6d41335ad53f1e94903 (tests, documentation, etc.), but that would just slow this PR down. I'd like to get this PR in as quickly as possible. If we did the Flask approach in a separate PR we could spend some more time on it.

Reply:
However, we still need [the original `main.py` content](https://github.com/alphagov/notifications-govuk-alerts/pull/151/files#diff-b10564ab7d2c520cdd0243874879fb0a782862c3c902ab535faabe57d5a505e1) in order to run the app locally from disk.

Reply:
I think we're confusing two different things here:

The write-to-disk is going to be a very temporary thing which will only be used until we change the pipeline to deploy the app *and* run a `cf run-task` to upload the new content.

In the meantime, for the local development it's much less hacky to use flask to serve the content instead of relying on hacks around filetypes and hacks around the filesystem limitations (`index` instead of `alerts` because we already have an `alerts/` dir)

Having said that, I'm going to reinstate the flask approach as it's much cleaner and standardised than what we used to have and it serves the exact same purpose.

Reply:
Restored in 506fb74